geoportal/geoportailv3_geoportal/lib/esri_authentication.py

```diff
@@ -0,0 +1,117 @@
+from typing import Optional, Dict
+import json
+from datetime import datetime, timedelta
+import requests
+import urllib
+from collections import namedtuple
+
+
+class ESRITokenException(Exception):
+    pass
+
+
+ResultTuple = namedtuple('ResultTuple', ['data', 'content_type'])
```

Comment:
Since we are in python 3.8 (or 3.7). You could use a dataclass.

Reply:
not worth the effort, IMO
And not compatible with current use unpack tuple AND named access